File: Data/Repositories/PessoaRepository.cs
using Microsoft.EntityFrameworkCore;
using Rinha.Data.Context;
using Rinha.Entities;

namespace Rinha.Data.Repositories
{
    public class PessoaRepository : IPessoaRepository
    {
        private readonly DataContext _context;
        public PessoaRepository(DataContext context)
        {
            _context = context;
        }

        public async Task<Pessoa?> Find(Guid id)
        {
            return await _context.Pessoas.FirstOrDefaultAsync(p => p.Id == id);
        }

        public async Task<List<Pessoa>> Search(string search)
        {
            return await _context.Pessoas
            .FromSqlRaw($"SELECT * FROM pessoas where pessoas.apelido ilike '%{search}%' or pessoas.nome ilike '%{search}%' or pessoas.stack ilike '%{search}%' LIMIT 50;", search)
            .ToListAsync();
        }

        public async Task<Pessoa?> FindByNome(string nome)
        {
            return await _context.Pessoas.FirstOrDefaultAsync(p => p.Nome.Equals(nome));
        }

        public async Task<Pessoa> Save(Pessoa pessoa)
        {
            await _context.AddAsync(pessoa);
            await _context.SaveChangesAsync();

            return pessoa;
        }

        public async Task<int> Count()
        {
            return await _context.Pessoas.CountAsync();
        }
    }
}

File: node/src/PessoaRepository.js
export class PessoaRepository {
  pool = undefined;

  constructor(pool){
    this.pool = pool
  }


  async save(pessoa){
    const sql = 'INSERT INTO pessoas VALUES ($1, $2, $3, $4, $5);'
    const values = [pessoa.id, pessoa.apelido, pessoa.nome, pessoa.nascimento, pessoa.stack?.join(",")]
    await this.pool.query(sql, values);
  }

  async count(){
    const result = await this.pool.query(`SELECT COUNT(*) FROM pessoas;`);

    return result.rows[0].count;
  }

  async search(term){
    const sql = "SELECT * FROM pessoas where pessoas.apelido ilike $1 or pessoas.nome ilike $1 or pessoas.stack ilike $1 LIMIT 50;"
    const values = ['%'+term+'%']

    const result = await this.pool.query(sql, values);
    return(result.rows.map(pessoa => ({
      ...pessoa, 
      nascimento: pessoa.nascimento.toISOString().split('T')[0], 
      stack: pessoa.stack?.split(',') || null
    })));
  }
}